components/engine-camel/src/main/java/org/eclipse/dirigible/components/engine/camel/processor/CamelProcessor.java

```diff
@@ -73,7 +72,19 @@ private void removeAllRoutes() {
             context.getRouteController().stopAllRoutes();
             context.getRouteController().removeAllRoutes();
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new CamelProcessorException(e);
+        }
+    }
+
+    public Object invokeRoute(String routeId, Object payload, Map<String, Object> headers) {
+        try (FluentProducerTemplate producer = context.createFluentProducerTemplate();) {
+            return producer
+                    .withHeaders(headers)
+                    .withBody(payload)
+                    .to(routeId)
+                    .request();
+        } catch (IOException e) {
+            throw new CamelProcessorException("Could not create fluent producer");
```

Comment:
Do not swallow exceptions and change the message to `"Could not invoke route: " + routeId`

Reply:
Done